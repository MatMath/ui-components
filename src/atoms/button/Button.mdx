---
name: Button
menu: Components
---

import { Playground, Props } from 'docz';
import Button from './Button';

# Button

A Button that allows users to take actions, and make choices, with a single tap. It can be a either:

- `Contained (filled) button`
- `Text button`

## Properties

<Props of={Button} />

## Usage

### Default variant for the button is `contained`

<Playground style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-around' }}>
  <div>
    <Button type="danger">Danger Button</Button>
  </div>

<div>
  <Button type='dark'>Dark Button</Button>
</div>

<div>
  <Button type='light'>Light Button</Button>
</div>

<div>
  <Button type='secondary'>Secondary Button</Button>
</div>

  <div>
    <Button type="success">Success Button</Button>
  </div>
</Playground>

### You can have a text button by specifying `variant` property to `text`

<Playground style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-around' }}>
  <div>
    <Button type="danger" variant="text">Danger Button</Button>
  </div>

<div>
  <Button type='dark' variant='text'>
    Dark Button
  </Button>
</div>

<div>
  <Button type='light' variant='text'>
    Light Button
  </Button>
</div>

<div>
  <Button type='secondary' variant='text'>
    Secondary Button
  </Button>
</div>

  <div>
    <Button type="success" variant="text">Success Button</Button>
  </div>
</Playground>

### In disabled mode, buttons will behave as following

<Playground className="test" style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-around', flexFlow: 'wrap' }}>
<div style={{ display: 'flex', margin: '8px' }}>
  <div style={{margin: '8px'}}>
    <Button type="danger" disabled>Danger Button</Button>
  </div>

<div style={{ margin: '8px' }}>
  <Button type='dark' disabled>
    Dark Button
  </Button>
</div>

<div style={{ margin: '8px' }}>
  <Button type='light' disabled>
    Light Button
  </Button>
</div>

<div style={{ margin: '8px' }}>
  <Button type='secondary' disabled>
    Secondary Button
  </Button>
</div>

  <div style={{margin: '8px'}}>
    <Button type="success" disabled>Success Button</Button>
  </div>
</div>

<div style={{ display: 'flex', margin: '8px' }}>
  <div style={{margin: '8px'}}>
    <Button type="danger" variant="text" disabled>Danger Button</Button>
  </div>

<div style={{ margin: '8px' }}>
  <Button type='dark' variant='text' disabled>
    Dark Button
  </Button>
</div>

<div style={{ margin: '8px' }}>
  <Button type='light' variant='text' disabled>
    Light Button
  </Button>
</div>

<div style={{ margin: '8px' }}>
  <Button type='secondary' variant='text' disabled>
    Secondary Button
  </Button>
</div>

  <div style={{margin: '8px'}}>
    <Button type="success" variant="text" disabled>Success Button</Button>
  </div>
  </div>
</Playground>
