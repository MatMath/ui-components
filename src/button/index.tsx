import * as React from 'react'

const styles = require('./styles.css')

export interface Props {
  label: string
}

export const Button: React.StatelessComponent<Props> = (props): JSX.Element => {
  return <button className={styles.button} type="button">{props.label}</button>
}

