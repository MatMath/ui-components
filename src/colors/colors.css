:root {

/* -- DASH_GREEN -- */
--dash-green-00: #0E353D;
--dash-green-01: #3E5D64;
--dash-green-02: #6E868B;
--dash-green-03: #9FAEB1;
--dash-green-04: #CFD7D8;
--dash-green-05: #D9DFE0;
--dash-green-06: #F5F7F7;

/* -- DASH_DARKER_GREEN -- */
--dash-darker-green-00: #071A1E;

/* -- MID_GREEN -- */
--mid-green-00: #0E6476;
--mid-green-01: #3E8391;
--mid-green-02: #6EA2AD;
--mid-green-03: #9FC1C8;
--mid-green-04: #CFE0E4;
--mid-green-05: #D9E6E9;

/* -- PINK -- */
--pink-00: #FDB5AF;
--pink-01: #FDC4BF;
--pink-02: #FED3CF;
--pink-03: #FEE1DF;
--pink-04: #FFF0EF;
--pink-05: #FFF3F2;

/* -- ORANGE -- */
--orange-00: #FE5B33;
--orange-01: #FE7C5C;
--orange-02: #FE9D85;
--orange-03: #FFBDAD;
--orange-04: #FFDED6;
--orange-05: #FFE5DE;

/* -- FUNCTIONAL_RED -- */
--functional-red-00: #4E000D;
--functional-red-01: #8B0018;
--functional-red-02: #FF002D;
--functional-red-03: #FFA49F;

/* -- RED -- */
--red-00: #BF0A33;
--red-01: #CC3B5C;
--red-02: #D96C85;
--red-03: #E59DAD;
--red-04: #F2CED6;
--red-05: #F5D8DE;

/* -- GREY -- */
--grey-00: #615B57;
--grey-01: #999592;
--grey-02: #CCC7C2;
--grey-03: #D6D2CE;
--grey-04: #E0DDDA;
--grey-05: #EBE9E7;
--grey-06: #F5F4F3;

/* -- BLACK -- */
--black: #000000;

/* -- WHITE -- */
--white: #FFFFFF;

/* -- VALIDATOR_GREEN -- */
--validator-green: #20B422;

/* -- TRANSPARENT -- */
--transparent: transparent;
}
