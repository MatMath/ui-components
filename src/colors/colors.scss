/* -- DASH_GREEN -- */
$dash-green-00: #0e353d;
$dash-green-01: #3e5d64;
$dash-green-02: #6e868b;
$dash-green-03: #9faeb1;
$dash-green-04: #cfd7d8;
$dash-green-05: #d9dfe0;
$dash-green-06: #f5f7f7;

/* -- DASH_DARKER_GREEN -- */
$dash-darker-green-00: #071a1e;

/* -- MID_GREEN -- */
$mid-green-00: #0e6476;
$mid-green-01: #3e8391;
$mid-green-02: #6ea2ad;
$mid-green-03: #9fc1c8;
$mid-green-04: #cfe0e4;
$mid-green-05: #d9e6e9;

/* -- PINK -- */
$pink-00: #fdb5af;
$pink-01: #fdc4bf;
$pink-02: #fed3cf;
$pink-03: #fee1df;
$pink-04: #fff0ef;
$pink-05: #fff3f2;

/* -- ORANGE -- */
$orange-00: #fe5b33;
$orange-01: #fe7c5c;
$orange-02: #fe9d85;
$orange-03: #ffbdad;
$orange-04: #ffded6;
$orange-05: #ffe5de;

/* -- FUNCTIONAL_RED -- */
$functional-red-00: #4e000d;
$functional-red-01: #8b0018;
$functional-red-02: #ff002d;
$functional-red-03: #ffa49f;

/* -- RED -- */
$red-00: #bf0a33;
$red-01: #cc3b5c;
$red-02: #d96c85;
$red-03: #e59dad;
$red-04: #f2ced6;
$red-05: #f5d8de;

/* -- GREY -- */
$grey-00: #615b57;
$grey-01: #999592;
$grey-02: #ccc7c2;
$grey-03: #d6d2ce;
$grey-04: #e0ddda;
$grey-05: #ebe9e7;
$grey-06: #f5f4f3;

/* -- BLACK -- */
$black: #000;

/* -- WHITE -- */
$white: #fff;

/* -- VALIDATOR_GREEN -- */
$validator-green: #20b422;

/* -- ACCESSIBLE_VALIDATOR_GREEN -- */
$accessible-validator-green: #278637;

/* -- ACCESSIBLE_VALIDATOR_RED -- */
$accessible-validator-red: #e00635;

/* -- TRANSPARENT -- */
$transparent: transparent;
