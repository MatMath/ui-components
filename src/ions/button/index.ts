export { default as BaseButton } from './base/BaseButton';
export * from './themes';
