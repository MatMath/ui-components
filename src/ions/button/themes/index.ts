export { default as ButtonDanger } from './danger/ButtonDanger';
export { default as ButtonDark } from './dark/ButtonDark';
export { default as ButtonLight } from './light/ButtonLight';
export { default as ButtonSecondary } from './secondary/ButtonSecondary';
export { default as ButtonSuccess } from './success/ButtonSuccess';
