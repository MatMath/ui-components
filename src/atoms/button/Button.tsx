import * as React from 'react';
import { BaseButtonProps } from '@atoms/button/types';
import {
  ButtonDanger,
  ButtonDark,
  ButtonLight,
  ButtonSecondary,
  ButtonSuccess
} from '@ions';

export const ButtonTypes = {
  danger: ButtonDanger,
  secondary: ButtonSecondary,
  dark: ButtonDark,
  light: ButtonLight,
  success: ButtonSuccess
};

interface Props extends BaseButtonProps {
  type: keyof typeof ButtonTypes;
}

const Button: React.FunctionComponent<Props> = (props: Props): JSX.Element => {
  const renderButtonByType = ({ type, ...props }: Props): JSX.Element => {
    const ButtonName = ButtonTypes[type];
    if (!ButtonName) {
      throw new Error(
        'Unknown button type. Please refer to the documentation to see possible types'
      );
    }

    return <ButtonName {...props} />;
  };

  return renderButtonByType(props);
};

export default Button;
