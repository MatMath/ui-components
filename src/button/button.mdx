---
name: Button
route: '/'
---

import { Playground, Props } from 'docz';
import Button from './';

# Button

A simple demonstration button

<Props of={Button}></Props>

## Basic usage [default settings]

<Playground>
  <Button />
</Playground>

## With a custom title

<Playground>
  <Button title={'My custom title'}></Button>
</Playground>

## With a primary setting

<Playground>
  <Button primary={true} title={'I am pink now!'}></Button>
</Playground><Paste>
