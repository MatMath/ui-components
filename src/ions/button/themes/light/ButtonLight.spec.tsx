import * as React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { ButtonLight } from '@ions';
import { $light } from '@globals/colors';
import { ButtonVariant } from '@atoms/button/Button';

describe('<ButtonLight>', () => {
  describe('global render', () => {
    it('should render', () => {
      const ContainedButtonLight = mount(<ButtonLight />);
      const TextButtonLight = mount(<ButtonLight variant={ButtonVariant.text} />);

      expect(ContainedButtonLight).toMatchSnapshot();
      expect(TextButtonLight).toMatchSnapshot();
    });

    describe('Light style', () => {
      it('should have the correct style for contained buttons ', () => {
        const ContainedButtonLight = mount(<ButtonLight />);

        expect(ContainedButtonLight).toHaveStyleRule('color', $light[700]);
        expect(ContainedButtonLight).toHaveStyleRule(
          'background-color',
          $light[300]
        );
      });

      it('should have the correct style for text buttons ', () => {
        const TextButtonLight = mount(<ButtonLight variant={ButtonVariant.text} />);

        expect(TextButtonLight).toHaveStyleRule('color', $light[700]);
        expect(TextButtonLight).toHaveStyleRule(
          'background-color',
          'transparent'
        );
      });
    });
  });
});
