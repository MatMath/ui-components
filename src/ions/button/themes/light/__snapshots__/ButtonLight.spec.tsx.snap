// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLight> global render should render 1`] = `
.c1 {
  line-height: 1.75;
  font-family: 'Source Sans Pro',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
}

.c0 {
  color: #474747;
  background-color: #f6f6f6;
  opacity: 1;
  cursor: pointer;
  border: 1px solid transparent;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  border-radius: 4px;
  line-height: 1.75;
  padding: 6px 16px;
  min-width: 64px;
  font-family: 'Source Sans Pro',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: 0;
}

.c0:hover,
.c0:active {
  border: 1px solid #e7e7e7;
}

.c0:active {
  background-color: #e7e7e7;
  border-color: #e7e7e7;
  color: #474747;
}

.c0:disabled span {
  pointer-events: none;
}

<WrappedButton>
  <BaseButton
    theme={
      Object {
        "activeBackground": "#e7e7e7",
        "background": "#f6f6f6",
        "color": "#474747",
        "hoverBorder": "#e7e7e7",
      }
    }
  >
    <styled.button
      theme={
        Object {
          "activeBackground": "#e7e7e7",
          "background": "#f6f6f6",
          "color": "#474747",
          "hoverBorder": "#e7e7e7",
        }
      }
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "lastClassName": "hUHuEU",
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  background-color: ",
                [Function],
                ";
  opacity: ",
                [Function],
                "
  cursor: ",
                [Function],
                "
  border: 1px solid ",
                "transparent",
                ";
  transition: all 0.2s ease-in;
  border-radius: 4px;  
  line-height: 1.75;
  padding: 6px 16px;
  min-width: 64px;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  display: inline-flex;
  position: relative;
  align-items: center;
  justify-content: center;
  outline: 0;
  &:hover, &:active {
    border: 1px solid ",
                [Function],
                "
  }
  &:active {
    background-color: ",
                [Function],
                ";
    border-color: ",
                [Function],
                "
    color: ",
                [Function],
                ";
  }
  &:disabled span {
    pointer-events: none;
  }
",
              ],
            },
            "displayName": "styled.button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        theme={
          Object {
            "activeBackground": "#e7e7e7",
            "background": "#f6f6f6",
            "color": "#474747",
            "hoverBorder": "#e7e7e7",
          }
        }
      >
        <button
          className="c0"
        >
          <styled.span>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  line-height: 1.75;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="c1"
              >
                Default
              </span>
            </StyledComponent>
          </styled.span>
        </button>
      </StyledComponent>
    </styled.button>
  </BaseButton>
</WrappedButton>
`;

exports[`<ButtonLight> global render should render 2`] = `
.c1 {
  line-height: 1.75;
  font-family: 'Source Sans Pro',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
}

.c0 {
  color: #474747;
  background-color: transparent;
  opacity: 1;
  cursor: pointer;
  border: 1px solid transparent;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  border-radius: 4px;
  line-height: 1.75;
  padding: 6px 16px;
  min-width: 64px;
  font-family: 'Source Sans Pro',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: 0;
}

.c0:hover,
.c0:active {
  border: 1px solid #f6f6f6;
}

.c0:active {
  background-color: #e7e7e7;
  border-color: #e7e7e7;
  color: #474747;
}

.c0:disabled span {
  pointer-events: none;
}

<WrappedButton
  variant="text"
>
  <BaseButton
    theme={
      Object {
        "activeBackground": "#e7e7e7",
        "background": "transparent",
        "color": "#474747",
        "hoverBorder": "#f6f6f6",
      }
    }
  >
    <styled.button
      theme={
        Object {
          "activeBackground": "#e7e7e7",
          "background": "transparent",
          "color": "#474747",
          "hoverBorder": "#f6f6f6",
        }
      }
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  background-color: ",
                [Function],
                ";
  opacity: ",
                [Function],
                "
  cursor: ",
                [Function],
                "
  border: 1px solid ",
                "transparent",
                ";
  transition: all 0.2s ease-in;
  border-radius: 4px;  
  line-height: 1.75;
  padding: 6px 16px;
  min-width: 64px;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  display: inline-flex;
  position: relative;
  align-items: center;
  justify-content: center;
  outline: 0;
  &:hover, &:active {
    border: 1px solid ",
                [Function],
                "
  }
  &:active {
    background-color: ",
                [Function],
                ";
    border-color: ",
                [Function],
                "
    color: ",
                [Function],
                ";
  }
  &:disabled span {
    pointer-events: none;
  }
",
              ],
            },
            "displayName": "styled.button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        theme={
          Object {
            "activeBackground": "#e7e7e7",
            "background": "transparent",
            "color": "#474747",
            "hoverBorder": "#f6f6f6",
          }
        }
      >
        <button
          className="c0"
        >
          <styled.span>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  line-height: 1.75;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="c1"
              >
                Default
              </span>
            </StyledComponent>
          </styled.span>
        </button>
      </StyledComponent>
    </styled.button>
  </BaseButton>
</WrappedButton>
`;
